snitun/multiplexer/core.py

```diff
@@ -42,6 +42,12 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+# If the payload is larger than 8192, use writelines to write the payload
+# to the stream. In Python 3.11+, writelines is a zero-copy operation.
+# For small payloads, the overhead of writelines is higher than the
+# overhead of write, so we only use writelines for larger payloads.
```

Comment:
If we had multiple byte strings to join, it would also make sense to lower this is a bit. For our use case, the max we join is two (header+body)